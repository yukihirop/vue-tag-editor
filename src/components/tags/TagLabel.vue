<template>
  <span :class="[tagAreaClass, tagCustomClass]">
    <span
      ref="tagname">
      <label :class="tagContentClass">{{ tagname }}</label>
    </span>
    <tag-delete-button
      :tagname="tagname"
      :event-hub="eventHub"
      :deleteAreaClass="deleteAreaClass"
      :deleteContentClass="deleteContentClass"
      @delete-tag="emitDeleteTag"
    />
  </span>
</template>

<script>
import TagDeleteButton from './TagDeleteButton'

export default {
  name: "TagLabel",
  components: {
    TagDeleteButton: TagDeleteButton
  },
  props:{
    tagname: {
      type: String,
      default: ''
    },
    eventHub: {
      type: Object,
      default(){
        return null
      }
    },
    tagAreaClass: {
      type: String,
      default: ""
    },
    tagContentClass: {
      type: String,
      default: ""
    },
    deleteAreaClass: {
      type: String,
      default: ""
    },
    deleteContentClass: {
      type: String,
      default: ""
    },
    tagCustomClass: {
      type: String,
      default: ""
    }
  },
  methods: {
    emitDeleteTag(){
      this.$emit('delete-tag')
    }
  }
}
</script>