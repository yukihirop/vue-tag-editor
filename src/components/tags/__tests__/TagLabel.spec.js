import { createLocalVue, mount } from '@vue/test-utils'
import TagLabel from '../TagLabel'

const localVue = createLocalVue()

describe('TagLabel', () => {
  const wrapper = mount(TagLabel, {
    localVue,
    propsData: {
      tagname: 'test'
    }
  })

  it('correct name options', () => {
    expect(wrapper.vm.$options.name).toBe('TagLabel')
  })

  it('renders correctly (snapshots)', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the correct markup', () => {
    expect(wrapper.html()).toContain("<span><span><label>test</label></span>")
    expect(wrapper.html()).toContain("x")
  })

  it('has a delete button', () => {
    expect(wrapper.contains('button')).toBe(true)
  })

  it('delete button click should call emitDeleteTag', () => {
    const button = wrapper.find('button')
    button.trigger('click')

    expect(wrapper.emitted('delete-tag')).toBeTruthy()
    expect(wrapper.emitted('delete-tag').length).toBe(1)
  })
})