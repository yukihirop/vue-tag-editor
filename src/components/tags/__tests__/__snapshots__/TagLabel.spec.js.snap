// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagLabel renders correctly (snapshots) 1`] = `
<span>
  <span>
    <label>
      test
    </label>
  </span>
   
  <button>
    
    x
  
  </button>
</span>
`;
